### DISCUSSION TITLE
"npm run example android" is throwing lockfile errors

### DISCUSSION THREAD

**Discussion by sadhan46** [callstack/react-native-builder-bob#655] [Q&A]
### Description

 Tried to run "npm run example android" and "yarn example android " for running my android app of react native but it threw an error of
"Internal Error: rn-ocr1-sdk@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile"

so I entered the command "yarn install" and still the yarn.lock file is not generated which according to internet has a workspace in it

I'm new to react native side as i'm try to create a sdk of flutter to trigger from react native side through android files

this where my error logs

sadhan@Sadhans-MacBook-Pro rn-ocr1-sdk % yarn example android

Internal Error: rn-ocr1-sdk@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
at V0.getCandidates (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:435:5145)
at kf.getCandidates (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:391:1264)
at /Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:439:8033
at df (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:390:11070)
at ge (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:439:8013)
at async Promise.allSettled (index 0)
at async io (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:390:10398)
at async /Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:439:8545
at async ti.startProgressPromise (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:390:47579)
at async je.resolveEverything (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:439:6573)

sadhan@Sadhans-MacBook-Pro rn-ocr1-sdk % yarn install
➤ YN0000: ┌ Resolution step
➤ YN0061: │ rimraf@npm:3.0.2 is deprecated: Rimraf versions prior to v4 are no longer supported
➤ YN0061: │ glob@npm:8.1.0 is deprecated: Glob versions prior to v9 are no longer supported
➤ YN0061: │ glob@npm:7.2.3 is deprecated: Glob versions prior to v9 are no longer supported
➤ YN0001: │ RequestError: read ECONNRESET
at ClientRequest. (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:195:14340)
at Object.onceWrapper (node:events:635:26)
at ClientRequest.emit (node:events:532:35)
at o.emit (/Users/sadhan/Desktop/rn-ocr1-sdk/.yarn/releases/yarn-3.6.1.cjs:190:90286)
at TLSSocket.socketErrorListener (node:_http_client:502:9)
at TLSSocket.emit (node:events:520:28)
at emitErrorNT (node:internal/streams/destroy:170:8)
at emitErrorCloseNT (node:internal/streams/destroy:129:3)
at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
at TLSWrap.onStreamRead (node:internal/stream_base_commons:218:20)
➤ YN0000: └ Completed in 4m 27s
➤ YN0000: Failed with errors in 4m 27s
➤ YN0061: inflight@npm:1.0.6 is deprecated: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
➤ YN0061: @babel/plugin-proposal-optional-chaining@npm:7.21.0 is deprecated: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
➤ YN0061: @babel/plugin-proposal-class-properties@npm:7.18.6 is deprecated: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
➤ YN0061: rimraf@npm:2.6.3 is deprecated: Rimraf versions prior to v4 are no longer supported
➤ YN0061: @babel/plugin-proposal-nullish-coalescing-operator@npm:7.18.6 is deprecated: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.

To reproduce
step 1). npx create-react-native-library@latest rn-login-sdk
step 2). cd rn-login-sdk
step 3). yarn example android

### Packages

- [X] create-react-native-library
- [ ] react-native-builder-bob

### Selected options

Turbo Native Modules with backward compatible
Kotlin & Objective-C
Vanilla

### Link to repro

https://github.com/sadhan46/rn-ocr1-sdk/tree/main

### Environment

System:
  OS: macOS 14.6.1
  CPU: (8) arm64 Apple M2
  Memory: 104.88 MB / 8.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.3.0
    path: /opt/homebrew/bin/node
  Yarn:
    version: 3.6.1
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.8.1
    path: /opt/homebrew/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.10121639
  Xcode:
    version: 14.3.1/14E300c
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.3
    wanted: 0.75.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: false
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

**Comment by mariojose96** [comment]
When building a new project with `npx create-react-native-library` you need to do the following steps to initialise your project:

1. cd into the project folder (in your case 'rn-login-sdk')
2. Run `yarn`
3. Run `yarn example android`

For further information about the development process follow the "CONTRIBUTING.md" instructions in the project root folder.

**Reply by atlj** [reply]
Yes, as @mariojose96 said, please take a look at the contributing file that's generated alongside your project. I'm converting this to a discussion since it doesn't seem like an issue with create-react-native-library itself.
